zeppelin-interpreter/src/main/java/org/apache/zeppelin/interpreter/remote/RemoteInterpreterServer.java

```diff
@@ -111,7 +113,8 @@ public void shutdown() throws TException {
     // this case, need to force kill the process
 
     long startTime = System.currentTimeMillis();
-    while (System.currentTimeMillis() - startTime < 2000 && server.isServing()) {
+    while (System.currentTimeMillis() - startTime < DEFAULT_SHUTDOWN_TIMEOUT &&
```

Comment:
While not really part of the scope of this PR, it is a chance to rework the logic to remove some unnecessary arithmetic and improve readability a bit.
Change `long startTime = System.currentTimeMillis();` to `long deadline = System.currentTimeMillis() + DEFAULT_SHUTDOWN_TIMEOUT;`
Change `while (System.currentTimeMillis() < deadline) &&`